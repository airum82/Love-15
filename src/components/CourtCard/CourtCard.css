.court-card {
  background-color: rgba(229, 149, 0, .7);
}