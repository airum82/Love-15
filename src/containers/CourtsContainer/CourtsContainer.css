.courts-container {
  display: grid;
  grid-template-columns: 25% 25% 25%;
  justify-content: space-around;
  grid-row-gap: 15px;
  margin-top: 5%;
}

.single-card {
  background-color: rgba(229, 149, 0, 1);
  width: 80%;
  margin: 0 auto
}

.single-card p {
  display: block;
}