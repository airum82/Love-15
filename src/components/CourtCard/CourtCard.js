import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './CourtCard.css';

export const CourtCard = ({name, location, id, addFavoriteCourt, account }) => {
  const court = {
    name,
    location
  }
  return (
    <div className="court-card">
      <h2>{name}</h2>
      <NavLink to={`/court/${id}`}>
        <p>{location}</p>
      </NavLink>
      <button onClick={() => addFavoriteCourt(account.id, court)}>
        add to favorites
      </button>
    </div>
  )
}