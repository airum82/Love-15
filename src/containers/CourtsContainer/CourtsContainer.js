import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { CourtCard } from '../../components/CourtCard/CourtCard';
import { fetchCourts } from '../../actions';

export class CourtsContainer extends Component {
  makeMapKey = (name) => {
    const makeMapKey = this.props.closeCourts.map(court => {
      if(court.name === name) {
        return {...court, map: true};
      }
        return court;
    });
    this.props.handleSubmitCourts(makeMapKey);
  }

  makeCourts = () => {
    let key = 0;
    return this.props.closeCourts.map(court => {
      return (
        <CourtCard
          name={court.name}
          location={court.location}
          makeMapKey={this.makeMapKey}
          key={key++}
        />
      )
    })
  }

  render() {
    if(this.props.closeCourts.length) {
      return (
        <div className="courts-container">
          {this.makeCourts()}
        </div>
      )
    } else {
      return (
        <div className="courts-container">
        </div>
      )
    }
  }
}

export const mapStateToProps = (state) => ({
  closeCourts: state.closeCourts
})

export const mapDispatchToProps = (dispatch) => ({
  handleSubmitCourts: (courts) => dispatch(fetchCourts(courts))
})

export default connect(mapStateToProps, mapDispatchToProps)(CourtsContainer)

CourtsContainer.propTypes = {
  closeCourts: PropTypes.array
}

