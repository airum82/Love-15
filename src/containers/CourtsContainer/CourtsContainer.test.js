import React from 'react';
import { shallow } from 'enzyme';
import { CourtsContainer } from './CourtsContainer';
import { CourtCard } from '../../components/CourtCard/CourtCard';

describe('CourtsContainer', () => {

  it('should match snapshot upon initial render', () => {
    const wrapper = shallow(<CourtsContainer closeCourts={[]}/>)

    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot upon receiving closeCourts props', () => {
    const mockCourts = [{}, {}];
    const wrapper = shallow(<CourtsContainer closeCourts={mockCourts} />);
    
    expect(wrapper).toMatchSnapshot();
  })

  it('makeMapKey should call handleSubmitCourts with correct params', () => {
    const mockCourts = [{ name: 'bob'}, {}];
    const mockHandleSubmitCourts = jest.fn();
    const expectedParams = [{ name: 'bob', map: true }, {}];
    const wrapper = shallow(<CourtsContainer
      closeCourts={mockCourts}
      handleSubmitCourts={mockHandleSubmitCourts} />
    );
    wrapper.instance().makeMapKey('bob');
    expect(mockHandleSubmitCourts).toHaveBeenCalledWith(expectedParams);
  })

  it('makeCourts should return an array of court components', () => {
    const mockMakeMapKey = jest.fn()
    const mockCourts = [
      { name: 'jim' , location: 'over there'}, 
      { name: 'sarah', location: 'not here'}
    ];
    const wrapper = shallow(<CourtsContainer
      closeCourts={mockCourts}/>);
    const expectedOutput = [
      <CourtCard
        key={0}
        name={'jim'}
        location={'over there'}
        makeMapKey={wrapper.instance().makeMapKey}
      />,
      <CourtCard
        key={1}
        name={'sarah'}
        location={'not here'}
        makeMapKey={wrapper.instance().makeMapKey}
      />
    ];
    expect(wrapper.instance().makeCourts()).toEqual(expectedOutput);
    
  })

})