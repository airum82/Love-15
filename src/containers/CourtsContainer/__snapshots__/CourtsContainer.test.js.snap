// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourtsContainer should match snapshot upon initial render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourtsContainer
    closeCourts={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "courts-container",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "courts-container",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CourtsContainer should match snapshot upon receiving closeCourts props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourtsContainer
    closeCourts={
        Array [
            Object {},
            Object {},
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          location={undefined}
          makeMapKey={[Function]}
          name={undefined}
/>,
        <Unknown
          location={undefined}
          makeMapKey={[Function]}
          name={undefined}
/>,
      ],
      "className": "courts-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "location": undefined,
          "makeMapKey": [Function],
          "name": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "location": undefined,
          "makeMapKey": [Function],
          "name": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            location={undefined}
            makeMapKey={[Function]}
            name={undefined}
/>,
          <Unknown
            location={undefined}
            makeMapKey={[Function]}
            name={undefined}
/>,
        ],
        "className": "courts-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "location": undefined,
            "makeMapKey": [Function],
            "name": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "location": undefined,
            "makeMapKey": [Function],
            "name": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
